### DISCUSSION TITLE
What data is stored in the database?

### DISCUSSION THREAD

**Discussion by MoritzLost** [allinurl/goaccess#2453] [Q&A] [answered]
I'm trying to find out exactly what data goaccess stores in the database with the `--persist` option. Background is that we need to comply with data privacy rules (in particular, deletion periods for private data). Are the entire log files stored, including IP addresses and User-Agents? Or does the database include only aggregated data?

If the database includes IP addresses and/or User-Agent strings, those would fall under the data privacy rules (GDPR). If only aggregated data is stored, with no personally identifiable information (PII), those rules wouldn't apply.

I'm not finding anything on this topic on the website or in the FAQ, can someone help me out? Thanks!

**Comment by allinurl** [comment, marked as answer]
It does store IPs and user agents, however, there's no correlation between the stored entries, only aggregated data — unless you pass `-a`. i.e., it will store the IPs in one table, user-agents in another and the only correlation between each table is the number of hits, visitors, TX and T.S. If you pass `-a`, it will link each IP and the corresponding user-agent.

Please let me know if that helps answer your question.

**Reply by MoritzLost** [reply]
@allinurl Thanks! This does help a lot. The data privacy laws probably still apply, since a single IP address without associated information is already considered 'private data', but we'll see what the data security officer has to say about that … thanks for the explanation!
